### PR TITLE
Addition of new Datasets

### PR DESCRIPTION
**Pull request by yk-csdev**
Added 3 CSV files for datasets
made some changes in main.py

### CODE REVIEW

File: backend/app/api/localdatasets.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import logging
+import csv
+from typing import List
+
+DATA_PATH='app/data/'
+output_max=10
+def get_unesco_sites(gpe: str):
```
Comment (bartekleon) [inline backend/app/api/localdatasets.py:7]: Probably entire code could be simplified.

```py
from typing import Any

def get_data_from_csv(filename: str, gpe: str, row_func: Callable[[Any], str]):
     with open(f"{DATA_PATH}{filename}.csv", 'r') as csvfile:
        reader = csv.DictReader(csvfile, delimiter=';')
        data_list: list[str] = []
        for row in reader:
            res = row_func(row, gpe)
            if res is not None:
               data_list.append(res)
            if len(data_list) >= output_max:
               break
        return data_list
```

And then you'd have for example:

```py
def _get_historical_places(row, gpe: str):
   if row.get('County', "").lower() == gpe.lower():
         return row['Resource_Name']
   return None

def get_historical_places(gpe: str):
   return get_data_from_csv('US_Historic_Places', gpe, _get_historical_places)
```
Or could use lambda function

File: backend/app/main.py [hunk 2]
```diff
@@ -66,11 +66,17 @@ def process_query(query: QueryRequest):
             return {"result": f"Prioritized event: {pinned_events}"}
 
         # Step 3: Fetch events from OpenTripMap
-        event_results = query_opentripmap(OpenTripMapModel(placename=city, kinds=keywords))
```
Comment (bartekleon) [inline backend/app/main.py:69]: Why have you made a change here?

File: backend/app/main.py [hunk 3]
```diff
@@ -66,11 +66,17 @@ def process_query(query: QueryRequest):
...
+        event_results = query_opentripmap(city, kinds=keywords)
 
         # Step 4: Fetch weather from OpenWeatherMap
         weather_info = weather.query_weather(city, date)
         
+        # Extract available datas from given GPE
+        if city and city.strip():  
+            unesco_sites = localdatasets.get_unesco_sites(city)
```
Comment (bartekleon) [inline backend/app/main.py:76]: keep style cohessive please. You made 3 different spacings for 3 calls.

File: backend/app/api/localdatasets.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
+import logging
+import csv
+from typing import List
```
Comment (bartekleon) [inline backend/app/api/localdatasets.py:3]: I don't see you using the `List` so probably can remove it, also it's preferable to just use `list` (lowercase), which you don't need to import

File: backend/app/main.py [hunk 5]
```diff
@@ -95,7 +104,7 @@ def process_query(query: QueryRequest):
                 message             = ans
             )
         )
-        return ans
+        return ans
```
Comment (bartekleon) [inline backend/app/main.py:107]: why a change here?

**Review by bartekleon — CHANGES_REQUESTED** [review]

File: backend/app/api/localdatasets.py (hunk 4 above)
Comment (yk-csdev) [reply backend/app/api/localdatasets.py:8]: Removed

File: backend/app/main.py (hunk 2 above)
Comment (yk-csdev) [reply backend/app/main.py:69]: Thats due to merge issue, resolved now

File: backend/app/main.py (hunk 3 above)
Comment (yk-csdev) [reply backend/app/main.py:76]: maintained the format as above

File: backend/app/main.py (hunk 5 above)
Comment (yk-csdev) [reply backend/app/main.py]: extra space, removed now

File: backend/app/main.py (hunk 2 above)
Comment (bartekleon) [reply backend/app/main.py:69]: Not addressed

File: backend/app/api/localdatasets.py (hunk 4 above)
Comment (bartekleon) [reply backend/app/api/localdatasets.py:8]: not addressed. Remove `from typing import List`

File: backend/app/api/localdatasets.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    - Implementation of extracting the local data available against the GPE provided
+"""
+import csv
+from typing import List
+
+DATA_PATH = 'app/data/'
+#To limit the output max to 10
+OUTPUT_MAX = 10
+
+def get_data(file_name: str, key: str, value: str, result_key: str) -> List[str]:
```
Comment (bartekleon) [inline backend/app/api/localdatasets.py:14]: use `list` instead of `List`

File: backend/app/api/localdatasets.py [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    with open(file_path, 'r') as csvfile:
+        reader = csv.DictReader(csvfile, delimiter=';')
+        for row in reader:
+            if key in row and row[key].lower() == value.lower():
+                data_list.append(row[result_key])
+                if len(data_list) >= OUTPUT_MAX:
+                    break
+    return data_list
+
+def get_unesco_sites(gpe: str) -> List[str]:
```
Comment (bartekleon) [inline backend/app/api/localdatasets.py:27]: `list[str]` not `List[str]`

File: backend/app/api/localdatasets.py [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    return get_data('unesco_sites', 'Country', gpe, 'Site Name')
+
+def get_historical_places(gpe: str) -> List[str]:
```
Comment (bartekleon) [inline backend/app/api/localdatasets.py:30]: same as above

File: backend/app/api/localdatasets.py [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+    return get_data('US_Historic_Places', 'County', gpe, 'Resource_Name')
+
+def get_hotels_motels(gpe: str) -> List[str]:
```
Comment (bartekleon) [inline backend/app/api/localdatasets.py:33]: same as above

**Review by bartekleon — CHANGES_REQUESTED** [review]

File: backend/app/api/localdatasets.py (hunk 4 above)
Comment (yk-csdev) [reply backend/app/api/localdatasets.py]: Changed to type list

**Pull request closed without merging**